webpack.config.js

```diff
@@ -1,35 +1,11 @@
-var path         = require( 'path' ),
-	SOURCE_DIR   = 'src/',
-	mediaEntries  = {},
-	mediaBuilds  = [ 'audiovideo', 'grid', 'models', 'views' ],
-	webpack      = require( 'webpack' );
-
-mediaBuilds.forEach( function ( build ) {
-	var path = SOURCE_DIR + 'wp-includes/js/media';
-	mediaEntries[ build ] = './' + path + '/' + build + '.manifest.js';
-} );
+const mediaConfig = require( './tools/webpack/media' );
+const packagesConfig = require( './tools/webpack/packages' );
 
 module.exports = function( env = { environment: "production" } ) {
-	const mode = env.environment;
-
-	const mediaConfig = {
-		mode,
-		cache: true,
-		entry: mediaEntries,
-		output: {
-			path: path.join( __dirname, 'src/wp-includes/js' ),
-			filename: 'media-[name].js'
-		},
-		optimization: {
-			// The files are minified by uglify afterwards. We could change this
-			// later, but for now prevent doing the work twice.
-			minimize: false
-		}
-	};
-
-	if ( mode === 'development' ) {
-		mediaConfig.watch = true;
-	}
+	const config = [
+		mediaConfig( env, __dirname ),
+		packagesConfig( env ),
+	];
 
-	return mediaConfig;
+	return config;
```

Comment:
Is this array of multiple configs something webpack supports by default? I wasn't aware of it :)
Ideally, media could be just another package but I can see how this may be needed for some time.

Reply:
I completely agree, I chose this approach pragmatically because I didn't want to touch the media config if possible.